src/m365/spo/commands/file/file-sharinglink-remove.spec.ts

```diff
@@ -0,0 +1,228 @@
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if the webUrl option is not a valid SharePoint site URL', async () => {
+    const actual = await command.validate({ options: { webUrl: 'foo', fileId: fileId, id: id } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the webUrl option is a valid SharePoint site URL', async () => {
+    const actual = await command.validate({ options: { webUrl: webUrl, fileId: fileId, id: id } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation if the fileId option is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { webUrl: webUrl, fileId: '12345', id: id } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('prompts before removing the specified sharing link to a file when confirm option not passed', async () => {
+    await command.action(logger, {
+      options: {
+        webUrl: webUrl,
+        fileId: fileId,
+        id: id
+      }
+    });
+
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the specified sharing link to a file when confirm option not passed and prompt not confirmed', async () => {
+    const postSpy = sinon.spy(request, 'delete');
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: false }
+    ));
+
+    await command.action(logger, {
+      options: {
+        webUrl: webUrl,
+        fileUrl: fileUrl,
+        id: id
+      }
+    });
+
+    assert(postSpy.notCalled);
+  });
+
+  it('removes specified sharing link to a file by fileId when prompt confirmed', async () => {
+    let sharingLinkRemoveCallIssued = false;
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${webUrl}/_api/web/GetFileById('${fileId}')?$select=SiteId,VroomItemId,VroomDriveId`) {
+        return fileInformationResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/${fileInformationResponse.SiteId}/drives/${fileInformationResponse.VroomDriveID}/items/${fileInformationResponse.VroomItemID}/permissions/${id}`) {
+        sharingLinkRemoveCallIssued = true;
+        return { statusCode: 204 };
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        verbose: true,
+        webUrl: webUrl,
+        fileId: fileId,
+        id: id
+      }
+    });
+    assert(sharingLinkRemoveCallIssued);
```

Comment:
Instead of using the `sharingLinkRemoveCallIssued`, we could put the stub on line 152 in a variable.

```
const deleteStub = sinon.stub(request, 'delete').callsFake(() => ............

....
assert(deleteStub.called)
```

This way we achieve the same, if the request URL in the delete stub doesn't match, the test will fail anyway because an exception is thrown on line 158.